ament_clang_tidy/ament_clang_tidy/main.py

```diff
@@ -165,8 +165,10 @@ def is_unittest_source(package, file_path):
         def start_subprocess(full_cmd):
             output = ''
             try:
-                output = subprocess.check_output(full_cmd,
-                                                 stderr=subprocess.DEVNULL).strip().decode()
+                output = subprocess.check_output(
+                    full_cmd,
+                    stderr=subprocess.DEVNULL
+                ).decode()
```

Comment:
Removing the `strip()` from the sequence prevent this:

```
/home/william/ros2_ws/src/ros2/rcutils/src/array_list.c:190:3: note: Call to function 'memcpy' is insecure as it does not provide security checks introduced in the C11 standard. Replace with analogous functions that support length arguments or provides boundary checks such as 'memcpy_s' in case of C11/home/william/ros2_ws/src/ros2/rcutils/src/char_array.c:123:5: warning: Call to function 'memcpy' is insecure as it does not provide security checks introduced in the C11 standard. Replace with analogous functions that support length arguments or provides boundary checks such as 'memcpy_s' in case of C11 [clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling]
    memcpy(char_array->buffer, old_buf, n);
    ^
```

Which was caused by a missing new line between files, it now looks correct:

```
/home/william/ros2_ws/src/ros2/rcutils/src/array_list.c:190:3: note: Call to function 'memcpy' is insecure as it does not provide security checks introduced in the C11 standard. Replace with analogous functions that support length arguments or provides boundary checks such as 'memcpy_s' in case of C11
/home/william/ros2_ws/src/ros2/rcutils/src/char_array.c:123:5: warning: Call to function 'memcpy' is insecure as it does not provide security checks introduced in the C11 standard. Replace with analogous functions that support length arguments or provides boundary checks such as 'memcpy_s' in case of C11 [clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling]
    memcpy(char_array->buffer, old_buf, n);
    ^
```